infrastructure/constructs/bundled_code.py

```diff
@@ -1,14 +1,73 @@
-from os.path import join
+from pathlib import Path
+from re import sub
+from subprocess import PIPE, check_call, run
+from sys import executable
 
 from aws_cdk import BundlingOptions, aws_lambda
 
 from .backend import BACKEND_DIRECTORY
 from .lambda_config import PYTHON_RUNTIME
 
 
-def bundled_code(directory: str) -> aws_lambda.Code:
+def poetry_export_extras(packaging_directory: str, lambda_directory: str) -> None:
+    # There isn't an elegant way of getting poetry to install package dependencies in a bespoke
+    # target lambda_directory within Python, so we export a requirements file and install using pip.
+    # This has been raised and discussed by the community as below:
+    # https://github.com/python-poetry/poetry/issues/1937
+
+    export_extras = run(
```

Comment:
It would be nice to not have to call a poetry subprocess. While it is possible to do `poetry add poetry` and call `import poetry` from within, I have yet to find any documentation or example around this on the Internet. Like `pip` (documentation below), I suspect this is a bad idea. Any code change within `poetry` could make things difficult to troubleshoot...